src/migrate/index.js

```diff
@@ -0,0 +1,59 @@
+const fs = require('node:fs');
+const path = require('node:path');
+const { Project } = require('ts-morph');
+
+const MIGRATION_DIR = `${__dirname}/migrations`;
+
+/**
+ * Lists all the available migrations.
+ *
+ * @returns {string[]} Array of migration names.
+ */
+function listMigrations() {
+  return fs
+    .readdirSync(MIGRATION_DIR)
+    .filter((fname) => fname.endsWith('.js'))
+    .map((fname) => fname.slice(0, -3));
+}
+
+/**
+ * @typedef {Object} Options
+ * @property {boolean} [isVerbose] - Optional flag for verbosity.
+ */
+
+/**
+ * Runs the migration specified by name with given options.
+ *
+ * @param {string} name - The name of the migration.
+ * @param {Options} options - Options for the migration.
+ * @returns {Promise<void>} A Promise that resolves when the migration is complete.
+ */
+async function runMigration(name, options) {
+  const { isVerbose } = options;
+
+  // runMigration is called by a CLI we want the directory
+  // the command is ran in and not the directory of this file
+  const tsconfigPath = path.join(process.cwd(), './tsconfig.json');
+  if (isVerbose) {
+    console.log(`Using the following tsconfig.json file: ${tsconfigPath}`);
+  }
+  const project = new Project({
+    tsConfigFilePath: path.join(tsconfigPath),
+  });
```

Comment:
I really like this feature of ts-morph when compared to jscodeshift, it can use a tsconfig file and then call `project.getSourceFiles()`. From memory when we migrated to typescript we had to pass file paths explicitly to jscodeshift